packages/web-pkg/src/components/Modals/SaveAsModal.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <div class="oc-height-1-1" tabindex="0">
+    <app-loading-spinner v-if="isLoading" />
+    <iframe
+      v-show="!isLoading"
+      ref="iframeRef"
+      class="oc-width-1-1 oc-height-1-1"
+      :title="iframeTitle"
+      :src="iframeSrc"
+      tabindex="0"
+      @load="onLoad"
+    ></iframe>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, onBeforeUnmount, onMounted, PropType, ref } from 'vue'
+import {
+  embedModeLocationPickMessageData,
+  Modal,
+  useClientService,
+  useGetMatchingSpace,
+  useMessages,
+  useModals,
+  useRouter,
+  useThemeStore
+} from '../../composables'
+import { RouteLocationRaw } from 'vue-router'
+import AppLoadingSpinner from '../AppLoadingSpinner.vue'
+import { Resource } from '@ownclouders/web-client'
+import { unref } from 'vue'
+import { resolveFileNameDuplicate } from '../../helpers'
+import { useGettext } from 'vue3-gettext'
+
+export default defineComponent({
+  name: 'SaveAsModal',
+  components: { AppLoadingSpinner },
+  props: {
+    modal: { type: Object as PropType<Modal>, required: true },
+    parentFolderLink: { type: Object as PropType<RouteLocationRaw>, required: true },
+    originalResource: { type: Object as PropType<Resource>, required: true },
+    content: { type: String, required: true }
+  },
+  setup(props) {
+    const iframeRef = ref<HTMLIFrameElement>()
+    const isLoading = ref(true)
+    const themeStore = useThemeStore()
+    const { $gettext } = useGettext()
+    const router = useRouter()
+    const { webdav } = useClientService()
```

Comment:
```suggestion
    const client = useClientService()
```

and later on `client.webdav`, otherwise the access token loses its reactivity.